modules/simple_forms_api/app/services/simple_forms_api/form_remediation/s3_client.rb

```diff
@@ -115,7 +115,7 @@ def generate_presigned_url(type: upload_type)
       def s3_upload_file_path(type)
         extension = File.extname(archive_path)
         ext = type == :submission ? '.pdf' : '.zip'
-        build_path(:file, s3_directory_path, archive_path, ext: extension ? nil : ext)
+        build_path(:file, s3_directory_path, File.basename(archive_path), ext: extension ? nil : ext)
```

Comment:
I had to read this a couple times to understand but I think it makes sense now. If the `archive_path` has an extension, we keep it, otherwise we use `.pdf` or `.zip` depending on `type`. Totally optional, but it might be more clearly written a bit more verbosely w/o the ternaries. But it's totally a question of taste!

Reply:
Yeah, ternaries can be so clean but can also obfuscate the logic. I also see I used a double ternary which isn't great either. I'll keep this in mind and perhaps throw in a refactor when I add more test coverage.